src/components/Sidebar/BranchTransition.tsx

```diff
@@ -1,22 +1,75 @@
-import React, { FC, memo } from 'react';
+import React, { FC, memo, useState, useCallback, ChangeEvent } from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faPlus, faTools } from '@fortawesome/free-solid-svg-icons';
-
+import DropDown from 'components/elements/DropDown';
+import { TextField, FormControl } from '@material-ui/core';
+import { setTransitionCondition, setTransitionConditionDescription } from 'utils/builder';
 import { SidebarHeader, SidebarButton } from '.';
 import { Pathway, Transition } from 'pathways-model';
+import { useCriteriaContext } from 'components/CriteriaProvider';
 import useStyles from './styles';
 
 interface BranchTransitionProps {
   pathway: Pathway;
+  currentNodeKey: string;
   transition: Transition;
   updatePathway: (pathway: Pathway) => void;
 }
 
-const BranchTransition: FC<BranchTransitionProps> = ({ pathway, transition, updatePathway }) => {
+const BranchTransition: FC<BranchTransitionProps> = ({
+  pathway,
+  currentNodeKey,
+  transition,
+  updatePathway
+}) => {
+  const { criteria } = useCriteriaContext();
+  const criteriaOptions = criteria.map(c => ({ value: c.label, label: c.label }));
   const styles = useStyles();
   const transitionKey = transition?.transition || '';
   const transitionNode = pathway.states[transitionKey];
+  const [useCriteriaSelected, setUseCriteriaSelected] = useState<boolean>(false);
+
+  const handleUseCriteria = useCallback((): void => {
+    setUseCriteriaSelected(!useCriteriaSelected);
+  }, [useCriteriaSelected]);
+
+  const selectCriteriaSource = useCallback(
+    (event: ChangeEvent<{ value: string }>): void => {
+      if (!currentNodeKey || !transition.id) return;
+
+      const criteriaSource = event?.target.value || '';
+      let elm = undefined;
+      criteria.forEach(c => {
+        if (c.label === criteriaSource) {
+          elm = c.elm;
+        }
+      });
```

Comment:
See comment above about using id
```suggestion
      criteria.forEach(c => {
        if (c.id === criteriaSource) {
          elm = c.elm;
        }
      });
```